packages/checkbox/src/index.js

```diff
@@ -0,0 +1,266 @@
+////////////////////////////////////////////////////////////////////////////////
+// Welcome to @reach/checkbox!
+
+import React, {
+  forwardRef,
+  createContext,
+  useContext,
+  useEffect,
+  useRef,
+  useState
+} from "react";
+import { wrapEvent, assignRef } from "@reach/utils";
+import warning from "warning";
+import { any, string, bool, func, oneOfType, node, shape } from "prop-types";
+
+const CustomCheckboxContext = createContext({});
+const useCheckboxContext = () => useContext(CustomCheckboxContext);
+
+////////////////////////////////////////////////////////////////////////////////
+export const CustomCheckboxContainer = forwardRef(
+  function CustomCheckboxContainer({ children, ...props }, forwardedRef) {
+    const [
+      { checked, mixed, disabled, readOnly, focused },
+      setContainerState
+    ] = useState({
+      checked: null,
+      mixed: null,
+      disabled: null,
+      readOnly: null,
+      focused: null
+    });
+
+    return (
+      <CustomCheckboxContext.Provider
+        value={{ setContainerState, visuallyHidden: true }}
+      >
+        <div
+          ref={forwardedRef}
+          {...props}
+          data-reach-custom-checkbox-container=""
+          data-checked={checked ? "" : undefined}
+          data-focus={focused ? "" : undefined}
+          data-mixed={mixed ? "" : undefined}
+          data-disabled={disabled ? "" : undefined}
+          data-read-only={readOnly ? "" : undefined}
+        >
+          {children}
+        </div>
+      </CustomCheckboxContext.Provider>
+    );
+  }
+);
+
+////////////////////////////////////////////////////////////////////////////////
+export const CustomCheckbox = forwardRef(function CustomCheckbox(
+  {
+    checked: controlledChecked,
+    defaultChecked,
+    disabled: disabledProp,
+    checkmarks,
+    children,
+    id,
```

Comment:
any reason we're messing around with `id`, `name` here? They just get forwarded to the underlying input.

Reply:
The rest of the props are forwarded to the wrapper, but even for a high-level `CustomCheckbox` abstraction users will regularly need to pass `id`, `name` and `value` to the input